public/app/features/explore/Explore.tsx

```diff
@@ -182,6 +182,30 @@ export class Explore extends React.PureComponent<Props, ExploreState> {
     }
   };
 
+  inspectQuery = async (check: string, key: string, value: string) => {
+    let isActive = false;
+    const { queries } = this.props;
+    for (const query of queries) {
+      const ds = await getDataSourceSrv().get(query.datasource);
+      if (ds.inspectQuery) {
+        isActive = ds.inspectQuery(query, {
+          check,
+          attributes: {
+            key,
+            value,
+          },
+        });
+      } else {
+        return false;
+      }
+    }
+    return isActive;
+  };
+
+  isFilterLabelActive = async (key: string, value: string) => {
+    return await this.inspectQuery('HAS_FILTER', key, value);
```

Comment:
Would be nice to have an enum with all possible values (at the moment only one), and some documentation what the values mean and how they correspond to ADD_FILTER / ADD_FILTER_OUT.